src/installer/managed/Microsoft.NET.HostModel/AppHost/ElfUtils.cs

```diff
@@ -1,51 +1,34 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.IO;
 
 namespace Microsoft.NET.HostModel.AppHost
 {
     internal static class ElfUtils
     {
-        // The Linux Headers are copied from elf.h
+        // First four bytes of valid ELF, as defined in https://github.com/torvalds/linux/blob/aae703b/include/uapi/linux/elf.h
+        //    0x7f (DEL), 'E', 'L', 'F'
+        private static ReadOnlySpan<byte> ElfMagic => "\u007f"u8 + "ELF"u8;
 
-#pragma warning disable 0649
-        private struct ElfHeader
+        public static bool IsElfImage(string filePath)
         {
-            private byte EI_MAG0;
-            private byte EI_MAG1;
-            private byte EI_MAG2;
-            private byte EI_MAG3;
+            using FileStream fileStream = File.OpenRead(filePath);
+            using BinaryReader reader = new(fileStream);
 
-            public bool IsValid()
+            if (reader.BaseStream.Length < 16) // EI_NIDENT = 16
             {
-                return EI_MAG0 == 0x7f &&
-                       EI_MAG1 == 0x45 &&
-                       EI_MAG2 == 0x4C &&
-                       EI_MAG3 == 0x46;
+                return false;
             }
 
+            byte[] eIdent = reader.ReadBytes(4);
```

Comment:
Is this .NET 7 project? You can remove the `BinaryReader` and just use `.ReadExactly` on the stream.

Reply:
It’s a netstandard2.0 library, so we can’t use new APIs (new language syntax is fine).